### PR TITLE
Fewer rhymebrain calls

### PR DESCRIPTION
**Pull request by mathildemouw**
I made a few updates to this script while I was working with it, to avoid getting rate-limited as frequently by RhymeBrain. Feel free to merge the changes in if you think it would be useful!

Some notes:

- I added a new variable `rhymingWordLists` because I didn't want to change the meaning of `rhymes` in case you were intending it to be used for something later
- I moved the group finding behavior into its own function to make it easier for me to isolate the call to RhymeBrain
- I removed out a stray console.log along my way

### CODE REVIEW

File: sketch.js [hunk 1]
```diff
@@ -150,53 +149,92 @@ function showRhymesOnly() {
...
+      var wColor = color(random(100,240), random(100,240), random(100,240));
+      // create UI for the results as they roll in
+      var span = createSpan(words[index]);
+      span.parent(output);
 
-        // Take all rhymes with an accuracy score over 200
-        for (var k = 0; k < data.length; k++) {
-          if (data[k]["score"] > 200) cRhymes.push(data[k]["word"]);
+      // First looks through our previous RhymeBrain responses to see if we already have the answer
+      for (var k = 0; k < rhymingWordLists.length; k++){
```
Comment (panzer) [inline sketch.js:204]: I'm getting the following error when I run this in my browser:
Screen Shot 2020-02-19 at 8 40 41 PM

File: sketch.js [hunk 2]
```diff
@@ -150,53 +149,92 @@ function showRhymesOnly() {
...
+        if (group.indexOf(word) != -1) {
+          var ci = groups.indexOf(group);
+          span.style('background-color', colors[ci]);
+        }
+      }
+    }
+  }
+}
+
+function findGroup(cRhymes, wColor) {
```
Comment (panzer) [inline sketch.js:171]: Thanks for splitting out these functions

File: sketch.js (hunk 1 above)
Comment (panzer) [reply sketch.js:204]: Is this just missing a `var rhymingWordLists =[];` at the top of the file?

**Review by panzer — COMMENTED** [review]
Hi, thanks so much for contributing!

I tried running your code in my browser but got an error. See below. Once that is fixed I'm excited to merge this!

File: sketch.js (hunk 1 above)
Comment (mathildemouw) [reply sketch.js:204]: Oops, I have it locally but it looks like I missed it when I picked my changes. I'll push up the fix shortly

Comment (mathildemouw) [reply sketch.js:204]: Alright, should be all good now

**Review by panzer — APPROVED** [review]
Thanks! 🎉

**Pull request merged**